### ISSUE TITLE
centos 8 complilation error

### ISSUE THREAD

**Issue by jeganvik** [shellinabox/shellinabox#485]
OS: centos 8
shellinabox : latest version 2.20

I try to install shellinabox, while I am compiling the code, I am getting below error 

ERROR:

make  all-am
make[1]: Entering directory '/tmp/shellinabox'
/bin/sh ./libtool  --tag=CC   --mode=link gcc -g -std=gnu99 -Wall -Os -static  -o shellinaboxd shellinabox/shellinaboxd.o shellinabox/externalfile.o shellinabox/launcher.o shellinabox/privileges.o shellinabox/service.o shellinabox/session.o shellinabox/usercss.o liblogging.la libhttp.la -lz -ldl -lutil 
libtool: link: gcc -g -std=gnu99 -Wall -Os -o shellinaboxd shellinabox/shellinaboxd.o shellinabox/externalfile.o shellinabox/launcher.o shellinabox/privileges.o shellinabox/service.o shellinabox/session.o shellinabox/usercss.o  ./.libs/liblogging.a ./.libs/libhttp.a -lz -ldl -lutil
./.libs/libhttp.a(ssl.o): In function `sslMakeContext':
/tmp/shellinabox/libhttp/ssl.c:642: undefined reference to `SSL_CTX_set_options'
./.libs/libhttp.a(ssl.o): In function `sk_SSL_COMP_zero':
/usr/include/openssl/ssl.h:964: undefined reference to `OPENSSL_sk_zero'
./.libs/libhttp.a(ssl.o): In function `sslMakeContext':
/tmp/shellinabox/libhttp/ssl.c:657: undefined reference to `SSL_CTX_set_options'
/tmp/shellinabox/libhttp/ssl.c:658: undefined reference to `SSL_CTX_set_options'
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:707: shellinaboxd] Error 1
make[1]: Leaving directory '/tmp/shellinabox'
make: *** [Makefile:540: all] Error 2

**Comment by genkimind** [comment]
+1 Fedora 27 docker image

**Comment by gbu-wafaux-integ** [comment]
as mentioned in issue 393 - https://github.com/shellinabox/shellinabox/issues/393 - a workaround-that-isnt-a-real-workaround would be to

> sudo apt-get install libssl1.0-dev

**Comment by hostingrajamani** [comment]
tried the same with cent os, the following is the result.

yum install libssl1.0-dev
Last metadata expiration check: 0:56:57 ago on Sat 27 Jun 2020 01:46:52 PM IST.
No match for argument: libssl1.0-dev
Error: Unable to find a match: libssl1.0-dev

 yum install libssl1.0
Last metadata expiration check: 0:59:31 ago on Sat 27 Jun 2020 01:46:52 PM IST.
No match for argument: libssl1.0
Error: Unable to find a match: libssl1.0

[root@vps shellinabox]# yum install libssl
Last metadata expiration check: 0:59:38 ago on Sat 27 Jun 2020 01:46:52 PM IST.
No match for argument: libssl
Error: Unable to find a match: libssl
[root@vps shellinabox]#

**Comment by xuguozhen** [comment]
Hi,all
I ran into the same problem on centos8.
Does anyone solve this problem？

**Comment by akoshodi** [comment]
Still experiencing this problem on CentOS 8. Is there an alternative method of installing shellinabox on CentOS 8?

**Comment by turboproc** [comment]
Unfortunately same on Ubuntu 20.04.2 LTS

**Comment by fabriziopicconi** [comment]
this is my recipe:
autoreconf -i
./configure
sed -i 's/\(^LIBS.*\)/\1-lssl -lcrypto /' Makefile
make

Compilation ends without error in my CentOS8 system

**Comment by neozeed** [comment]
> -lssl -lcrypto

Yep, adding the libraries to the LIBS in the Makefile did the trick, thanks!

**Comment by yanndegat** [comment]
## shell in a box install                                                                                                                                                                                     
 (cd /tmp;                                                                                                                                                                                                     
  git clone https://github.com/shellinabox/shellinabox.git && cd shellinabox;                                                                                                                                  
 autoreconf -i; ./configure LIBS="-lssl -lcrypto" && make && make install;    
  cp etc-pam.d-shellinabox-example /etc/pam.d/shellinabox;                     
  cd /tmp; rm -Rf shellinabox )

**Comment by bicigringo** [comment]
Just made a small bash script to install Shell in a box from source on a Redhat based 8+ machine (Used Rocky) using standard locations :

```
dnf -y install git openssl-devel pam-devel zlib-devel autoconf automake libtool
cd /usr/src 
git clone https://github.com/shellinabox/shellinabox.git && cd shellinabox
autoreconf -i
# BUG https://github.com/shellinabox/shellinabox/issues/485
./configure LIBS="-lssl -lcrypto" && make && make install
cp etc-pam.d-shellinabox-example /etc/pam.d/shellinabox

cat <<=== > /usr/lib/systemd/system/shellinaboxd.service
[Unit]
Description=Shell In A Box daemon
Documentation=man:shellinaboxd(1)
After=network.target nss-lookup.target

[Service]
EnvironmentFile=-/etc/sysconfig/shellinaboxd
WorkingDirectory=/usr/share/shellinabox
ExecStart=/usr/sbin/shellinaboxd -u $USER -g $GROUP --cert=${CERTDIR} --port=${PORT} $OPTS
Restart=on-failure

[Install]
WantedBy=multi-user.target
===

cat <<=== > /etc/sysconfig/shellinaboxd
# Shell in a box daemon configuration
# For details see shellinaboxd man page

# Basic options
USER=shellinabox
GROUP=shellinabox
CERTDIR=/var/lib/shellinabox
PORT=4200
OPTS="--disable-ssl -s /:LOGIN --localhost-only"

# Additional examples with custom options:

# Fancy configuration with right-click menu choice for black-on-white:
# OPTS="--user-css Normal:+black-on-white.css,Reverse:-white-on-black.css --disable-ssl-menu -s /:LOGIN"

# Simple configuration for running it as an SSH console with SSL disabled:
# OPTS="-t -s /:SSH:host.example.com"
===

groupadd -g 950 shellinabox
useradd -u 950 -g 950 shellinabox -s /sbin/nologin
mkdir /var/lib/shellinabox 
chown shellinabox: /var/lib/shellinabox
ln -s /usr/local/bin/shellinaboxd /usr/sbin/shellinaboxd

mkdir -p /usr/share/shellinabox
cp /usr/src/shellinabox/shellinabox/*css /usr/share/shellinabox

systemctl daemon-reload
systemctl enable --now shellinaboxd
```
